clients/admin-ui/src/features/privacy-experience/Preview.tsx

```diff
@@ -0,0 +1,260 @@
+import { useToast } from "@fidesui/react";
+import {
+  CookieKeyConsent,
+  EmptyExperience,
+  FidesOptions,
+  PrivacyExperience,
+  UserGeolocation,
+} from "fides-js/src/lib/consent-types";
+import { useFormikContext } from "formik";
+import Script from "next/script";
+import React, { useEffect, useState } from "react";
+
+import { getErrorMessage } from "~/features/common/helpers";
+import { TranslationWithLanguageName } from "~/features/privacy-experience/form/helpers";
+import { useLazyGetPrivacyNoticeByIdQuery } from "~/features/privacy-notices/privacy-notices.slice";
+import {
+  ComponentType,
+  ExperienceConfigCreate,
+  ExperienceTranslation,
+  LimitedPrivacyNoticeResponseSchema,
+  PrivacyNoticeResponse,
+  SupportedLanguage,
+} from "~/types/api";
+
+export type Fides = {
+  consent: CookieKeyConsent;
+  experience?: PrivacyExperience | EmptyExperience;
+  geolocation?: UserGeolocation;
+  options: FidesOptions;
+  init: (config: any) => {};
+};
+
+declare global {
+  interface Window {
+    Fides: Fides;
+  }
+}
+
+enum FidesPreviewComponent {
+  BANNER = "banner",
+  MODAL = "modal",
+}
+
+const Preview = ({
+  allPrivacyNotices,
+  initialValues,
+  translation,
+  isMobilePreview,
+}: {
+  allPrivacyNotices: Partial<LimitedPrivacyNoticeResponseSchema[]>;
+  initialValues: Partial<ExperienceConfigCreate>;
+  translation?: TranslationWithLanguageName;
+  isMobilePreview: boolean;
+}) => {
+  const { values } = useFormikContext<ExperienceConfigCreate>();
+  const [noticesOnConfig, setNoticesOnConfig] = useState<
+    PrivacyNoticeResponse[]
+  >([]);
+
+  const toast = useToast();
+
+  const [getPrivacyNoticeByIdTrigger] = useLazyGetPrivacyNoticeByIdQuery();
+
+  const getPrivacyNotice = async (id: string) => {
+    const result = await getPrivacyNoticeByIdTrigger(id);
+    if (result.isError) {
+      const errorMsg = getErrorMessage(
+        result.error,
+        "A problem occurred while fetching privacy notice data.  Some notices may not display correctly on the preview."
+      );
+      toast({ status: "error", description: errorMsg });
+    }
+    const { data } = await getPrivacyNoticeByIdTrigger(id);
+    return data;
+  };
+
+  useEffect(() => {
```

Comment:
Technically according to best practices, this shouldn't be in a useEffect-- it only depends on a single React state variable, so ideally we'd: 
* track the notices we care about in state in the component above this
* whenever the ids are changed in the form, change it via an event handler
* pass it as a prop into this component
That said, I tried doing it that way and it introduced absolutely horrendous lag, so it lives here.

Reply:
@jpople just so I'm aware, what's the downside of using `useEffect` here when it depends on a single React state variable?

Reply:
@eastandwestwind You know what, I think I'm mistaken here-- I was misremembering the principles from [You Might Not Need an Effect](https://react.dev/learn/you-might-not-need-an-effect).  The fact that this is fetching from the API means it's legitimate here, what I was thinking of is that it's better not to use it to *deterministically* calculate state from other state inside the app.